drop-storage/src/lib.rs

```diff
@@ -62,194 +63,290 @@ impl Storage {
+                    true
                 }
-                true
-            }
-            TransferFiles::Outgoing(files) => {
-                trace!(
-                    self.logger,
-                    "Inserting transfer::Outgoing files len {}",
-                    files.len()
-                );
-
-                for file in files {
-                    Self::insert_outgoing_path(&conn, transfer.id, file)?;
+                TransferFiles::Outgoing(files) => {
+                    trace!(
+                        self.logger,
+                        "Inserting transfer::Outgoing files len {}",
+                        files.len()
+                    );
+
+                    for file in files {
+                        Self::insert_outgoing_path(&self.logger, &conn, transfer.id, file);
+                    }
+                    false
                 }
-                false
-            }
-        };
+            };
+
+            sync::insert_transfer(&conn, transfer.id, is_incoming)?;
 
-        sync::insert_transfer(&conn, transfer.id, is_incoming)?;
+            conn.commit()?;
 
-        conn.commit()?;
+            Ok::<(), Error>(())
+        };
 
-        Ok(())
+        if let Err(e) = task().await {
+            error!(self.logger, "Failed to insert transfer"; "error" => %e);
+        }
     }
 
-    pub fn update_transfer_sync_states(
+    pub async fn update_transfer_sync_states(
         &self,
         transfer_id: Uuid,
         remote: Option<sync::TransferState>,
         local: Option<sync::TransferState>,
-    ) -> Result<()> {
-        let mut conn = self.conn.lock().expect("Failed to acquire a lock");
-        let conn = conn.transaction()?;
+    ) {
+        let task = || async {
+            let mut conn = self.conn.lock().await;
+            let conn = conn.transaction()?;
 
-        if let Some(remote) = remote {
-            sync::transfer_set_remote_state(&conn, transfer_id, remote)?;
-        }
-        if let Some(local) = local {
-            sync::transfer_set_local_state(&conn, transfer_id, local)?;
-        }
+            if let Some(remote) = remote {
+                sync::transfer_set_remote_state(&conn, transfer_id, remote)?;
+            }
+            if let Some(local) = local {
+                sync::transfer_set_local_state(&conn, transfer_id, local)?;
+            }
+
+            conn.commit()?;
 
-        conn.commit()?;
-        Ok(())
+            Ok::<(), Error>(())
+        };
+
+        if let Err(e) = task().await {
+            error!(self.logger, "Failed to update transfer sync states"; "error" => %e);
+        }
     }
 
-    pub fn transfer_sync_state(&self, transfer_id: Uuid) -> crate::Result<Option<sync::Transfer>> {
-        let conn = self.conn.lock().expect("Failed to acquire a lock");
-        sync::transfer_state(&conn, transfer_id)
+    pub async fn transfer_sync_state(&self, transfer_id: Uuid) -> Option<sync::Transfer> {
+        let task = || async {
```

Comment:
Same as above. The closure is not needed since the async blocks already encapsulates the task